src/inmanta/logging.py

```diff
@@ -54,6 +59,102 @@ def _is_on_tty() -> bool:
 }
 
 
+class LoggingConfigExtension:
+    """
+    The logging config defined by an extension
+    """
+
+    def __init__(
+        self,
+        formatters: Optional[dict[str, object]] = None,
+        handlers: Optional[dict[str, object]] = None,
+        loggers: Optional[dict[str, object]] = None,
+        root_handlers: Optional[set[str]] = None,
+        log_dirs_to_create: Optional[set[str]] = None,
```

Comment:
I wasn't planning to do anything like that. But we can. I'll make the change.

Reply:
I may be missing something, in which case this may not be required. If you weren't planning anything like this, what did you have in mind?

Reply:
Just read the dictConfig from file and pass it to `logging.config.dictConfig()`.

Reply:
I mean about the `log_dirs_to_create` option. How would a user indicate they need a certain directory to be created for their logs?

Reply:
Ah. I considered that one the responsibility of the user if a custom config was provided.  But we can also make it the responsibility of the server.

Reply:
I'm fine with leaving it up to the user, and perhaps prefer it even. I just assumed you wanted to include it in that interface, because you included it in this one.

The intermediate layer I proposed in my initial comment was purely to be able to support this "superset" rather than the exact Python format. So I don't think that's relevant, given this new information.